tests/5.0/teams_loop/test_target_teams_loop_allocate.c

```diff
@@ -0,0 +1,259 @@
+//===-------------- test_target_teams_loop_allocate.c---------------------===//
+//
+// OpenMP API Version 5.0 Nov 2018
+// The tests in this file tests 'teams loop' construct coupled with map()
+// clause in combination with different allocator types
+//
+//===----------------------------------------------------------------------===//
+
+#include <omp.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include "ompvv.h"
+
+#define N 1024
+
+/**
+  This is a basic test to demonstrate how allocate can be used with
+  target teams loop construct. Allocate clause is used to allocate a local
+  variable of size int.
+*/
+int testTargetTeamsAllocateCl() {
+  int a[N], b[N], c[N], local = 0;
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    c[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams map(to: a[0:N], b[0:N]) map(from: c[0:N]) \
+        uses_allocators(omp_cgroup_mem_alloc) \
+        allocate(omp_low_lat_mem_alloc: local) private(local)
+  #pragma omp loop private(local)
+  for (int i = 0; i < N; i++) {
+    local = a[i] + b[i];
+    c[i] = local;
+  }
+  // Validate
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, c[i] != a[i] + b[i]);
+  }
+  OMPVV_TEST_AND_SET_VERBOSE(errors, local != 0);
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate use of allocate clause with
+  omp_default_mem_alloc allocator to allocate an array.
+*/
+int testTargetTeamsAllocateArrayDefaultAlloc() {
+  int a[N], b[N], c[N], local[N];
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    local[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_default_mem_alloc)\
+        allocate(omp_default_mem_alloc: local) private(local)
+  for (int i = 0; i < N; i++) {
+    local[i] = a[i] + b[i];
+    c[i] = local[i];
+  }
+  // Validate
+  for (int i = 0; i < N; i++) {
+    OMPVV_TEST_AND_SET_VERBOSE(errors, c[i] != a[i] + b[i]);
+    OMPVV_TEST_AND_SET_VERBOSE(errors, local[i] != 0);
+  }
+  return errors;
+}
+
+/**
+  This is a basic test to demonstrate use of allocate clause with
+  omp_large_cap_mem_alloc allocator to allocate an array.
+*/
+int testTargetTeamsAllocateArrayLargeCapAlloc() {
+  int a[N], b[N], c[N], local[N];
+  int errors = 0;
+  // Data Inititalize
+  for (int i = 0; i < N; i++) {
+    a[i] = 2*i;  // Even
+    b[i] = 2*i + 1;  // Odd
+    local[i] = 0;
+  }
+  // Execute on target
+#pragma omp target teams loop map(to: a[0:N], b[0:N]) map(from: c[0:N])\
+        uses_allocators(omp_large_cap_mem_alloc)\
+        allocate(omp_large_cap_mem_alloc: local) private(local)
```

Comment:
The error @tob2 mentioned still exists here (`error: 'local' specified in 'allocate' clause but not in an explicit privatization clause`); one possible solution is to use `firstprivate(local)` instead of `private(local)`, but using the `firstprivate` clause will incur unnecessary initialization of the firstprivate variable, `local`.

Reply:
Replaced "private(local)" with  "firstprivate(local)" as suggested...